src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -687,11 +703,18 @@ bool GCToOSInterface::VirtualDecommit(void* address, size_t size)
 bool GCToOSInterface::VirtualReset(void * address, size_t size, bool unlock)
 {
     int st;
+    bool madviseFlags = MADV_FREE;
```

Comment:
this needs to be under `#if HAVE_MADV_FREE`

Reply:
Done. I've also simplified the code here too. Should be more obvious what happens in all the different conditions.